web/src/features/panels/zone/ZoneDetails.tsx

```diff
@@ -45,10 +44,10 @@ export default function ZoneDetails(): JSX.Element {
     }
     // When first hitting the map (or opening a zone from the ranking panel),
     // set the correct matching view mode (zone or country).
-    if (hasSubZones && isZoneView) {
+    if (!isSubZone) {
```

Comment:
I don't think this will work. Now it will always switch to country mode when clicking on a zone that don't have any subzones.

For example if you are in "zone mode" and click on Germany it will switch to "country mode".

We want to preserve the mode if the zone don't have any subzones at all and only switch if there are "two levels" to a zone.

Reply:
Okay, I've reintroduced the hasSubZone condition. Do you think this will cover all cases?

Reply:
Seems to work great!

If you remove that leftover console.log() we can merge this.